File: apple/Sources/ReactNativeWatchOS/Modifiers/GridCellColumns.swift
import SwiftUI

/// SwiftUI `.gridCellColumns(_:)`. Makes a view span multiple `Grid`
/// columns. Available on watchOS 9+.
enum RNWGridCellColumnsModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("gridCellColumns") { view, params, _ in
            let count = params.int("count") ?? 1
            return AnyView(view.gridCellColumns(count))
        }
    }
}

File: src/modifiers/gridCellColumns.ts
import { createModifier } from './createModifier';

export interface GridCellColumnsParams {
  /// Number of columns this cell should span.
  count: number;
}

/// SwiftUI `.gridCellColumns(_:)`. Tells the enclosing `Grid` that a view
/// spans `count` columns.
export function gridCellColumns(
  params: GridCellColumnsParams
): ReturnType<typeof createModifier>;
export function gridCellColumns(
  count: number
): ReturnType<typeof createModifier>;
export function gridCellColumns(a: GridCellColumnsParams | number) {
  if (typeof a === 'number') {
    return createModifier('gridCellColumns', { count: a });
  }
  return createModifier('gridCellColumns', a);
}
